plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -307,13 +185,14 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
             return Optional.empty();
         }
 
-        Optional<Table> hiveTable = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), name.getTableName());
-        if (hiveTable.isEmpty() || !isIcebergTable(hiveTable.get())) {
+        Table table;
```

Comment:
Add a comment here
```java
// load the base table for the system table
```
I had to look at `IcebergTableName` to make sure that was what this was doing